src/content-layout/index.tsx

```diff
@@ -2,29 +2,43 @@
 // SPDX-License-Identifier: Apache-2.0
 import React, { useContext, useRef } from 'react';
 import clsx from 'clsx';
-import { AppLayoutContext } from '../../../app-layout/visual-refresh/context';
+import { AppLayoutContext } from '../app-layout/visual-refresh/context';
+import { applyDisplayName } from '../internal/utils/apply-display-name';
 import { ContentLayoutProps } from './interfaces';
-import { useVisualRefresh } from '../../hooks/use-visual-mode';
+import { getBaseProps } from '../internal/base-component';
+import useBaseComponent from '../internal/hooks/use-base-component';
+import { useDynamicOverlap } from '../app-layout/visual-refresh/hooks/use-dynamic-overlap';
+import { useMergeRefs } from '../internal/hooks/use-merge-refs';
+import { useVisualRefresh } from '../internal/hooks/use-visual-mode';
 import styles from './styles.css.js';
-import { useDynamicOverlap } from '../../../app-layout/visual-refresh/hooks/use-dynamic-overlap';
 
 export { ContentLayoutProps };
 
-export default function ContentLayout({ children, disableOverlap, header }: ContentLayoutProps) {
+export default function ContentLayout({ children, disableOverlap, header, ...rest }: ContentLayoutProps) {
+  const baseProps = getBaseProps(rest);
   const { breadcrumbs } = useContext(AppLayoutContext);
+
   const rootElement = useRef<HTMLDivElement>(null);
+  const { __internalRootRef } = useBaseComponent('ContentLayout');
+  const mergedRef = useMergeRefs(rootElement, __internalRootRef);
+  const overlapElement = useDynamicOverlap();
   const isVisualRefresh = useVisualRefresh(rootElement);
-  const isOverlapDisabled = !children || !header || disableOverlap;
 
-  const overlapElement = useDynamicOverlap();
+  /**
+   * Disable the overlap if the component is missing either a header or child
+   * content. If the component is not using visual refresh then the overlap
+   * will not be displayed at all. This is handled in the CSS not the JavaScript.
+   */
+  const isOverlapDisabled = !children || !header || disableOverlap;
```

Comment:
Should this also effect the `useDynamicOverlap` hook?

Reply:
It shouldn't. Even if the overlap is disabled within the content layout component, it still needs to extend the background from the app layout component down to the overlap element.